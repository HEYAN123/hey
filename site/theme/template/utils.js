
import flattenDeep from "lodash/flattenDeep";
import flatten from "lodash/flatten";

// 获取本地网页路径
export function getLocalizedPathname(path, zhCN, hash) {
  const pathname = path.startsWith("/") ? path : `/${path}`;
  let fullPath;
  if (!zhCN) {
    // to enUS
    fullPath = /\/?index-cn/.test(pathname) ? "/" : pathname.replace("-cn", "");
  } else if (pathname === "/") {
    fullPath = "/index-cn";
  } else if (pathname.endsWith("/")) {
    fullPath = pathname.replace(/\/$/, "-cn/");
  } else {
    fullPath = `${pathname}-cn`;
  }

  if (hash) {
    const localHash = hash[zhCN ? "zhCN" : "enUS"];
    fullPath += `#${localHash}`;
  }
  return fullPath;
}

export function getMetaDescription(jml = []) {
  const COMMON_TAGS = ["h1", "h2", "h3", "p", "img", "a", "code", "strong"];
  if (!Array.isArray(jml)) {
    return "";
  }
  const paragraph = flattenDeep(
    jml
      .filter(item => {
        if (Array.isArray(item)) {
          const [tag] = item;
          return tag === "p";
        }
        return false;
      })
      // ['p', ['code', 'aa'], 'bb'] => ['p', 'aabb']
      .map(item => {
        const [tag, ...others] = flatten(item);
        const content = others
          .filter(other => typeof other === "string" && !COMMON_TAGS.includes(other))
          .join("");
        return [tag, content];
      }),
  ).find(p => p && typeof p === "string" && !COMMON_TAGS.includes(p));
  return paragraph;
}

export function isZhCN(pathname) {
  return /-cn\/?$/.test(pathname);
}
