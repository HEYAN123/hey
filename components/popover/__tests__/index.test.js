import React from 'react';
import { render, mount } from 'enzyme';
import Popover from '..';
import mountTest from '../../../tests/shared/mountTest';
import { sleep } from '../../../tests/utils';
import ConfigProvider from '../../config-provider';

describe('Popover', () => {
  mountTest(Popover);

  it('should show overlay when trigger is clicked', async () => {
    const popover = mount(
      <Popover content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(popover.instance().getPopupDomNode()).toBe(null);

    popover.find('span').simulate('click');
    await sleep(100);

    const popup = popover.instance().getPopupDomNode();
    expect(popup).not.toBe(null);
    expect(popup.className).toContain('ant-popover-placement-top');
    expect(popup.innerHTML).toMatchSnapshot();
    expect(popup.innerHTML).toMatchSnapshot();
  });

  it('props#overlay do not warn anymore', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});

    const overlay = jest.fn();
    mount(
      <Popover content="console.log('hello world')" title="code" trigger="click">
        <span>show me your code</span>
      </Popover>,
    );

    expect(errorSpy.mock.calls.length).toBe(0);
    expect(overlay).not.toHaveBeenCalled();
  });

  it(`should be rendered correctly in RTL direction`, () => {
    const wrapper = mount(
      <ConfigProvider direction="rtl">
        <Popover title="RTL" visible>
          <span>show me your Rtl demo</span>
        </Popover>
      </ConfigProvider>,
    );
    expect(render(wrapper)).toMatchSnapshot();
  });
});
