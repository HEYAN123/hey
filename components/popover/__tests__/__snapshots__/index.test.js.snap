// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover should be rendered correctly in RTL direction 1`] = `
Array [
  <span
    class="ant-popover-open"
  >
    show me your Rtl demo
  </span>,
  <div>
    <div
      class="ant-popover ant-popover-rtl"
      style="opacity:0"
    >
      <div
        class="ant-popover-content"
      >
        <div
          class="ant-popover-arrow"
        />
        <div
          class="ant-popover-inner"
          role="tooltip"
        >
          <div>
            <div
              class="ant-popover-title"
            >
              RTL
            </div>
            <div
              class="ant-popover-inner-content"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Popover should show overlay when trigger is clicked 1`] = `"<div class=\\"ant-popover-content\\"><div class=\\"ant-popover-arrow\\"></div><div class=\\"ant-popover-inner\\" role=\\"tooltip\\"><div><div class=\\"ant-popover-title\\">code</div><div class=\\"ant-popover-inner-content\\">console.log('hello world')</div></div></div></div>"`;

exports[`Popover should show overlay when trigger is clicked 2`] = `"<div class=\\"ant-popover-content\\"><div class=\\"ant-popover-arrow\\"></div><div class=\\"ant-popover-inner\\" role=\\"tooltip\\"><div><div class=\\"ant-popover-title\\">code</div><div class=\\"ant-popover-inner-content\\">console.log('hello world')</div></div></div></div>"`;
